File: 451-sort-characters-by-frequency/sort-characters-by-frequency.cpp
class Solution {
public:
    string frequencySort(string s) {
        map<char, int> freq;
        for (char c: s)
            freq[c]++;
        priority_queue<pair<int, char>> maxHeap;
        for (auto& entry: freq)
            maxHeap.push({entry.second, entry.first});
        string res;
        while (!maxHeap.empty()){
            if (maxHeap.top().first > 1){
                for (int j=0; j<maxHeap.top().first; j++)
                    res += maxHeap.top().second;
                maxHeap.pop();
            } else {
                res += maxHeap.top().second;
                maxHeap.pop();
            }
        }
        return res;
    }
};

File: 451-sort-characters-by-frequency/sort-characters-by-frequency.py
class Solution(object):
    def frequencySort(self, s):
        """
        :type s: str
        :rtype: str
        """
        freq = {}
        for i in s:
            if i in freq:
                freq[i] += 1
            else:
                freq[i]  = 1
        freq = sorted(freq.items(), key = lambda i : freq[i[0]] , reverse=True)
        str = ""
        for x in freq:
            if x[1] > 1:
                for y in range(x[1]):
                    str += x[0]
            else : 
                str += x[0]
        return str